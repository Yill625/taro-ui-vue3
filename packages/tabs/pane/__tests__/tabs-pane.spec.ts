import {
  genMountFn,
  testPropClassAndStyle
} from '@taro-ui-vue3/test-utils/helper'
import AtTabsPane from '../index.vue'

describe('AtTabsPane', () => {

  const mountFn = genMountFn(AtTabsPane)

  it('should render default and match snapshot', () => {
    const wrapper = mountFn({}, { default: ['test'] })
    expect(wrapper.element).toMatchSnapshot()
  })

  testPropClassAndStyle(mountFn)

  it.each([
    'vertical',
    'horizontal'
  ])('should render prop tabDirection -- %s', async (tabDirection) => {
    const wrapper = mountFn({ tabDirection })
    expect(
      wrapper.find('.at-tabs-pane--vertical').exists()
    ).toBe(tabDirection === 'vertical')
  })

  it('should render props index and current', async () => {
    const wrapper = mountFn()
    expect(
      wrapper.classes()
    ).toContain('at-tabs-pane--active')
    expect(
      wrapper.classes()
    ).not.toContain('at-tabs-pane--inactive')

    await wrapper.setProps({ index: 1, current: 0 })
    expect(
      wrapper.classes()
    ).not.toContain('at-tabs-pane--active')
    expect(
      wrapper.classes()
    ).toContain('at-tabs-pane--inactive')
  })
})