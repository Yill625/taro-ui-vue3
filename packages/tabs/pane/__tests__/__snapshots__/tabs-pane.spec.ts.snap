// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTabsPane should render default and match snapshot 1`] = `
<view
  class="at-tabs-pane--active at-tabs-pane"
>
  
  test
  
</view>
`;
